main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -902,7 +902,26 @@ object Typer extends Phase[ResolvedAst.Root, TypedAst.Root] {
 
       case ResolvedAst.Expression.Cast(exp, declaredTyp, declaredEff, tvar, loc) =>
```

Comment:
I think this can already be done in the Resolver. Is that not where we have the other checks?